src/vs/platform/terminal/common/terminal.ts

```diff
@@ -4,10 +4,26 @@
  *--------------------------------------------------------------------------------------------*/
 
 import { Event } from 'vs/base/common/event';
+import { IDisposable } from 'vs/base/common/lifecycle';
 import { IProcessEnvironment } from 'vs/base/common/platform';
 import { URI } from 'vs/base/common/uri';
 import { IGetTerminalLayoutInfoArgs, IPtyHostProcessReplayEvent, ISetTerminalLayoutInfoArgs } from 'vs/platform/terminal/common/terminalProcess';
 
+
+export interface IWindowsShellHelper extends IDisposable {
+	readonly onShellNameChange: Event<string>;
+	getShellType(title: string): TerminalShellType;
+	getShellName(): Promise<string>;
+}
```

Comment:
Let's just move this into windowsShellHelper.ts now since we only plan on using it inside TerminalProcess

Reply:
It's also referenced in [TerminalInstance](https://github.com/microsoft/vscode/blob/8a209fd731e8039bb1fc0f9030de842c4b2bed94/src/vs/workbench/contrib/terminal/browser/terminalInstance.ts#L102-L102) and if I move that, it's bad layering

Reply:
We can remove that now, it's just a member property and disposal of it. Speaking of which, we will need to make sure this still works:

https://github.com/microsoft/vscode/blob/d0a514a651e52fcb3c4ab1619fe85523f67bd0ee/src/vs/workbench/contrib/terminal/browser/terminalInstance.ts#L1485-L1492

When the terminal's title is set by the API (via actual API or Terminal: Rename command), we want to ignore all title events but keep all shell type events (existing bug that this is not handled right now).

Reply:
still works

Reply:
Cool, I guess that makes sense because it's using the non-Windows path now